pkg/operator/ceph/client/controller.go

```diff
@@ -285,36 +286,84 @@ func (r *ReconcileCephClient) createOrUpdateClient(cephClient *cephv1.CephClient
-	_, err = r.context.Clientset.CoreV1().Secrets(cephClient.Namespace).Get(r.clusterInfo.Context, secret.Name, metav1.GetOptions{})
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			logger.Debugf("creating secret for %q", secret.Name)
-			if _, err := r.context.Clientset.CoreV1().Secrets(cephClient.Namespace).Create(r.clusterInfo.Context, secret, metav1.CreateOptions{}); err != nil {
-				return errors.Wrapf(err, "failed to create secret for %q", secret.Name)
+	// Check if another controller owns it
+	if err == nil {
+		owner := metav1.GetControllerOf(existing)
+
+		if owner != nil && !checkCephClientIsOwner(*owner, cephClient) {
+			if cephClient.Spec.RemoveSecret {
+				// secret is owned by different cephClient returning from here
+				// log it
+				return nil
 			}
-			logger.Infof("created client %q", cephClient.Name)
+			return fmt.Errorf("secret %s/%s is already owned by %s %s",
+				secret.Namespace, secret.Name, owner.Kind, owner.Name)
+		}
+		// secret is owned by this ceph client. remove it if requied
+		if cephClient.Spec.RemoveSecret {
+			err := r.context.Clientset.CoreV1().
+				Secrets(secret.Namespace).
+				Delete(r.clusterInfo.Context, secret.Name, metav1.DeleteOptions{})
+			if err != nil && !kerrors.IsNotFound(err) {
+				return errors.Wrapf(err, "failed to delete secret %q", secret.Name)
+			}
+			logger.Infof("removed secret %q due to RemoveSecret=true", secret.Name)
 			return nil
 		}
-		return errors.Wrapf(err, "failed to get secret for %q", secret.Name)
 	}
-	logger.Debugf("updating secret for %s", secret.Name)
-	_, err = r.context.Clientset.CoreV1().Secrets(cephClient.Namespace).Update(r.clusterInfo.Context, secret, metav1.UpdateOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "failed to update secret for %q", secret.Name)
+
+	if err := controllerutil.SetControllerReference(cephClient, secret, r.scheme); err != nil {
+		return errors.Wrapf(err, "failed to set owner reference on secret %q", secret.Name)
+	}
+
+	if existing == nil {
+		logger.Debugf("creating secret %q", secret.Name)
+		if _, err := r.context.Clientset.CoreV1().
+			Secrets(secret.Namespace).Create(r.clusterInfo.Context, secret, metav1.CreateOptions{}); err != nil {
+			return errors.Wrapf(err, "failed to create secret %q", secret.Name)
+		}
+		logger.Infof("created secret for CephClient %q", cephClient.Name)
+	} else {
+		logger.Debugf("updating secret %q", secret.Name)
+		if _, err := r.context.Clientset.CoreV1().
+			Secrets(secret.Namespace).Update(r.clusterInfo.Context, secret, metav1.UpdateOptions{}); err != nil {
+			return errors.Wrapf(err, "failed to update secret %q", secret.Name)
+		}
+		logger.Infof("updated secret for CephClient %q", cephClient.Name)
 	}
 
-	logger.Infof("updated client %q", cephClient.Name)
 	return nil
 }
 
+// checkCephClientIsOwner true if a is the cephClient.
+func checkCephClientIsOwner(a metav1.OwnerReference, cephClient *cephv1.CephClient) bool {
+	aGV, err := schema.ParseGroupVersion(a.APIVersion)
```

Comment:
acronyms are hard to understand
```suggestion
	ownerGroup, err := schema.ParseGroupVersion(a.APIVersion)
```